### PR TITLE
[TASK] Moved the snippet preview call (for CMS9) and save scores call to backend context, refactoring and small cleanup

### PR DESCRIPTION
**Pull request by RinyVT**
## Summary

This PR can be summarized in the following changelog entry:

* Moved the snippet preview ajax call to the backend (only for CMS9, CMS8 still needs the frontend for URL generation)
* Moved the save scores ajax call to the backend for both CMS8 and CMS9
* Refactored some classes, bit of cleanup and function-splitups
* Moved the preview typoscript to CMS8.typoscript
* Deleted the yoast_seo_save_scores typoscript and Preview.typoscript file
* Introduced a new PreviewService which both the AjaxController and SnippetPreview userFunction use
* Moved standard labels to LocaleService
* Added the `TYPO3_SITE_PATH` constant to the `getUrlForType` function to fix the problem with installations on a subdomain

## Relevant technical choices:

* Prominent words url has not been changed because that requires a change in the premium package

## Test instructions

This PR can be tested by following these steps:

* Pull this branch within a CMS9 and CMS8 installation and test if the snippet preview still works (for pages and TCA records)
* See if this fixes the problem with an installation on subdomain, this is now only applicable to CMS8 since CMS9 does this through backend

## Quality assurance

* [X] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

### CODE REVIEW

File: Classes/Service/UrlService.php [hunk 1]
```diff
@@ -113,13 +130,28 @@ public function getRouteEnhancerError(): bool
...
     /**
      * @param int $type
+     * @param string $additionalGetParameters
      * @return string
      */
-    public function getUrlForType($type): string
+    public function getUrlForType($type, $additionalGetParameters = ''): string
     {
-        return '/?type=' . $type;
+        return GeneralUtility::getIndpEnv('TYPO3_SITE_PATH') . '?type=' . $type . $additionalGetParameters;
```
Comment (maechler) [inline Classes/Service/UrlService.php]: This does not work reliably in TYPO3 9, please see https://github.com/Yoast/Yoast-SEO-for-TYPO3/pull/323

Comment (haassie) [reply Classes/Service/UrlService.php]: This method is not used in v9. See version compare in line 72.

File: Configuration/TypoScript/Setup/CMS8.typoscript [hunk 2]
```diff
@@ -269,3 +269,27 @@ yoast_seo_sitemap {
...
 }
+
+yoast_seo_preview = PAGE
+yoast_seo_preview {
+    typeNum = {$config.yoast_seo.fe_preview_type}
+    config {
+        disableAllHeaderCode = 1
+        absRefPrefix = /
+
+        no_cache = 1
```
Comment (haassie) [inline Configuration/TypoScript/Setup/CMS8.typoscript]: can we add the no debug as well?

File: Classes/Service/UrlService.php (hunk 1 above)
Comment (maechler) [reply Classes/Service/UrlService.php]: Oh, I see. That looks good to me.

File: Configuration/TypoScript/setup.txt [hunk 3]
```diff
@@ -1,27 +1,3 @@
 <INCLUDE_TYPOSCRIPT: source="FILE:EXT:yoast_seo/Configuration/TypoScript/Setup/Config.typoscript">
 <INCLUDE_TYPOSCRIPT: source="FILE:EXT:yoast_seo/Configuration/TypoScript/Setup/Module.typoscript">
-<INCLUDE_TYPOSCRIPT: source="FILE:EXT:yoast_seo/Configuration/TypoScript/Setup/Preview.typoscript">
-
-yoast_seo_save_scores = PAGE
```
Comment (haassie) [inline Configuration/TypoScript/setup.txt:5]: You remove the save scores here. Is this moved to somewhere else?

File: Configuration/TypoScript/setup.txt [hunk 4]
```diff
@@ -1,27 +1,3 @@
...
-            20 {
-                header = Access-Control-Allow-Origin: *
-            }
-        }
-    }
-
-    10 = USER_INT
-    10.userFunc = YoastSeoForTypo3\YoastSeo\UserFunctions\SaveScores->render
-}
+<INCLUDE_TYPOSCRIPT: source="FILE:EXT:yoast_seo/Configuration/TypoScript/Setup/Preview.typoscript">
```
Comment (haassie) [inline Configuration/TypoScript/setup.txt:3]: You deleted this file in this PR. What was your intention?

File: Configuration/TypoScript/setup.txt (hunk 3 above)
Comment (RinyVT) [reply Configuration/TypoScript/setup.txt:5]: Yes, this has been moved to a backend route, see Configuration/Backend/AjaxRoutes.php and AjaxController::saveScoresAction

**Pull request merged**